qiskit/providers/aer/jobs/utils.py

```diff
@@ -55,35 +56,139 @@ def wrapper(*args, **kw):
     # Check if we don't need to split
     if max_size is None or not max_size > 0:
-        return qobj
+        return qobj, seed
+
     num_jobs = ceil(len(qobj.experiments) / max_size)
     if num_jobs == 1:
-        return qobj
+        return qobj, seed
 
+    qobjs = []
     # Check for parameterizations
     params = getattr(qobj.config, 'parameterizations', None)
-    qobjs = []
+
+    exp_id = None
+    shift_index = 0
+    seed_shift = 256
+    _seed = 0
     for i in range(num_jobs):
         sub_id = qobj_id or str(uuid.uuid4())
         indices = slice(i * max_size, (i + 1) * max_size)
         sub_exp = qobj.experiments[indices]
         sub_config = qobj.config
+
         if params is not None:
-            sub_config = copy.copy(qobj.config)
             sub_config.parameterizations = params[indices]
+            sub_config = copy.copy(qobj.config)
+
+        if seed > 0:
+            id_dat = getattr(sub_exp[0].header, "metadata", None)
+            if id_dat is not None and "id" in id_dat:
+                _id = id_dat["id"]
+                if _id == exp_id:
+                    shift_index = shift_index + 1
+                else:
+                    exp_id = _id
+                    shift_index = 0
+                _seed = seed + seed_shift * shift_index
+            if sub_config is qobj.config:
+                sub_config = copy.copy(qobj.config)
+            setattr(sub_config, "seed_simulator", _seed)
+
         qobjs.append(type(qobj)(sub_id, sub_config, sub_exp, qobj.header))
-    return qobjs
+
+    if seed > 0:
+        seed = _seed + seed_shift
+
+    return qobjs, seed
+
+
+def _check_custom_instruction(experiments, custom_instructions):
+    # check if custom instruction exist
+    for exp in experiments:
+        for inst in exp.instructions:
+            if inst.name in custom_instructions:
+                return True
+    return False
+
+
+def split_qobj(qobj, config, max_size=None, max_shot_size=None, qobj_id=None):
+    """Split a qobj and return a list of qobjs each with a single experiment.
+
+    Args:
+        qobj (Qobj): The input qobj object to split
+        config (BackendConfiguration): backend configuration.
+        max_size (int or None): the maximum number of circuits per job. If
+            None don't split (Default: None).
+        max_shot_size (int or None): the maximum number of shots per job. If
+            None don't split (Default: None).
+        qobj_id (str): Optional, set a fixed qobj ID for all subjob qobjs.
+
+    Raises:
+        JobError : If max_job_size > 1 and seed is set.
+        JobError : If custom instructions exist.
+
+    Returns:
+        List: A list of qobjs.
+    """
+    split_qobj_list = []
+    if (max_shot_size is not None and max_shot_size > 0):
+        if _check_custom_instruction(qobj.experiments, config.custom_instructions):
+            raise JobError("cluster backend does not support custom instructions.")
```

Comment:
This is a confusing message because there is no "cluster backend". This should refer to the experiment option value that doenst support this:
```suggestion
            raise JobError("`max_shot_size` option cannot be used with circuits containing save instructions.")
```
Also custom instructions like `Kraus` and `SuperOp` and `ReadoutError` support splitting. It should only be save instructions that prevent splitting so you should check for them.